src/orion/algo/hyperband.py

```diff
@@ -426,10 +485,22 @@ def observe(self, trials):
                 continue
 
     @property
-    def is_done(self):
-        """Return True, if all required execution been done."""
+    def is_done(self) -> bool:
+        """Return True, if all required execution has been done."""
         if self.executed_times >= self.repetitions:
             return True
+        fidelity_index = self.fidelity_index
```

Comment:
Perhaps this test could be in BaseAlgorithm. If there is a fidelity dimension, we take it into consideration, otherwise only count the trials. This should apply to any fidelity-based algorithm.

Reply:
Does every fidelity-based algorithm want this logic? Could a new fidelity-based algorithm want something different?
(Related question: Do we have any fidelity-based algorithms that aren't based on Hyperband? If so, do they need this?)

Reply:
Yes the logic would be valid for any fidelity-based algorithms. The idea is that as long as a trial is not a full fidelity than it is not considered completed and the algorithm has not exhausted the space.

PBT does not inherit from Hyperband. It would not apply for PBT but it overwrites BaseAlgorithm.is_done anyway.

Reply:
There's a little catch: Here we ignore any `max_trials`, but we don't in the BaseAlgorithm. How could we then use super().is_done here? See what I mean?

Reply:
Oh I see. Maybe we should break is_done down internally into 2 methods like _has_exhausted_cardinality and _has_reached_max_trials. BaseAlgo would use both in is_done, but child classes could reuse one or the other if necessary.

Reply:
Ok makes sense, I'll try something like that.

Reply:
Fixed in 1aedf7b34ff5cca1e282c778a2892e07634f6daa